### PR TITLE
chore(SPAR-112): add workflow to deploy spark web docs to s3

### PR DESCRIPTION
**Pull request by rickyrattlesnake**
## Description

Deploys the docs site to the static web server at [spark-web.staging.](https://spark-web.staging.cloud.brighte.com.au/)

Updates Docs links to the new site and adds README.

**NOTE:** The deployment doesn't work until we configure the github OIDC provider see ticket https://brighte.atlassian.net/browse/SPAR-112

## Associated Tickets

- SPAR-112

## Checklist

- [x] I've updated unit tests where needed
- [x] I've updated the components README.md where needed
- [x] I've added major version bumps for my breaking changes
- [x] I've added changesets where needed

### CODE REVIEW

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Deploy Docs
+on:
+  push:
+    branches: [storybook-deploy-2]
+  # workflow_run:
```
Comment (rickyrattlesnake) [inline .github/workflows/deploy-docs.yml:5]: uncomment once we configure permissions properly

File: .github/workflows/deploy-docs.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          node-version-file: .nvmrc
+          cache: yarn
+      - name: install dependencies
+        run: yarn install --immutable
+      - name: build docs
+        run: yarn build:docs
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: arn:aws:iam::1234567890:role/fake-role
```
Comment (darkbluesun) [inline .github/workflows/deploy-docs.yml:39]: question: what is fake role?

**Review by ChristopherMitchell242 — APPROVED** [review]

**Review by darkbluesun — COMMENTED** [review]
are we still waiting for the OIDC connector?

**Comment by darkbluesun** [comment]
Should we remove the vercel steps?

File: .github/workflows/deploy-docs.yml (hunk 2 above)
Comment (rickyrattlesnake) [reply .github/workflows/deploy-docs.yml]: We need to get the real one that's associated with the OIDC provider

**Comment by rickyrattlesnake** [comment]
Hehe - hello friends, should I close this PR @ChristopherMitchell242 🐞 ?

**Pull request closed without merging**